test/e2e/tests/custom-token-send-transfer.spec.js

```diff
@@ -0,0 +1,426 @@
+        });
+        await driver.clickElement({
+          text: 'Assets',
+          tag: 'button',
+        });
+        const assets = await driver.findElement('[title="8.5 TST"]');
+        assert.ok(assets, 'Token amount is not correct');
+      },
+    );
+  });
+
+  it('can be transferred from a dapp without specifying gas', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions,
+        smartContract,
+        title: this.test.title,
+        failOnConsoleError: false,
+      },
+      async ({ driver, contractRegistry }) => {
+        const contractAddress = await contractRegistry.getContractAddress(
+          smartContract,
+        );
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        // create token
+        await driver.openNewPage(
+          `http://127.0.0.1:8080/?contract=${contractAddress}`,
+        );
+        let windowHandles = await driver.getAllWindowHandles();
+        const extension = windowHandles[0];
+        await driver.findClickableElement('#deployButton');
+        // add token from dapp
+        await driver.clickElement({
+          text: 'Add Token to Wallet',
+          tag: 'button',
+        });
+
+        await driver.waitUntilXWindowHandles(3);
+        windowHandles = await driver.getAllWindowHandles();
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.clickElement({ text: 'Add token', tag: 'button' });
+        windowHandles = await getWindowHandles(driver, 2);
+        // transfer token without gas from dapp
+        await driver.clickElement({
+          text: 'Transfer Tokens Without Gas',
+          tag: 'button',
+        });
+
+        await driver.waitUntilXWindowHandles(3);
+        windowHandles = await driver.getAllWindowHandles();
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.waitForSelector({
+          tag: 'h1',
+          text: '1.5 TST',
+        });
+        await driver.clickElement({ text: 'Confirm', tag: 'button' });
+        // checks transaction finish correctly from extension
+        await driver.switchToWindow(extension);
+        await driver.clickElement({ tag: 'button', text: 'Activity' });
+        await driver.wait(async () => {
+          const pendingTxes = await driver.findElements(
+            '.transaction-list__pending-transactions .transaction-list-item',
+          );
+          return pendingTxes.length === 1;
+        }, 10000);
+
+        await driver.waitForSelector({
+          css: '.transaction-list-item__primary-currency',
+          text: '-1.5 TST',
+        });
```

Comment:
We should be able to remove the following code block, as the line that follows waits for the completed transaction list item
```
        await driver.wait(async () => {
          const pendingTxes = await driver.findElements(
            '.transaction-list__pending-transactions .transaction-list-item',
          );
          return pendingTxes.length === 1;
        }, 10000);

        await driver.waitForSelector({
          css: '.transaction-list-item__primary-currency',
          text: '-1.5 TST',
        });
```

Reply:
Done